service/pki/service.go

```diff
@@ -118,73 +118,6 @@ func (s *service) IsMounted(clusterID string) (bool, error) {
 	return true, nil
 }
 
-func (s *service) IsRoleCreated(clusterID string) (bool, error) {
-	// Create a client for the logical backend configured with the Vault token
-	// used for the current cluster's PKI backend.
-	logicalBackend := s.VaultClient.Logical()
-
-	// Check if a PKI for the given cluster ID exists.
-	secret, err := logicalBackend.List(s.ListRolesPath(clusterID))
-	if IsNoVaultHandlerDefined(err) {
-		return false, nil
-	} else if err != nil {
-		return false, microerror.Mask(err)
-	}
-
-	// In case there is not a single role for this PKI backend, secret is nil.
-	if secret == nil {
-		return false, nil
-	}
-
-	// When listing roles a list of role names is returned. Here we iterate over
-	// this list and if we find the desired role name, it means the role has
-	// already been created.
-	if keys, ok := secret.Data["keys"]; ok {
-		if list, ok := keys.([]interface{}); ok {
-			for _, k := range list {
-				if str, ok := k.(string); ok && str == s.RoleName(clusterID) {
-					return true, nil
-				}
-			}
-		}
-	}
-
-	return false, nil
-}
-
-func (s *service) VerifyPKISetup(clusterID string) (bool, error) {
```

Comment:
actually I dont think it needs to be put back.

In the cert operator code, we are calling pki/service#Create, which has it's own checking for the PKI Backend existing or not. So cert-operator is checking things twice.

From a design point of view, I also argue that currently, cert-operator is doing too much 'ask' and not enough 'tell'. It shouldn't know that it has to call this specific method before it can run the Create method.  (and it doesn't, because the Create method already guards itself)